endpoints/api_request.py

```diff
@@ -135,6 +135,48 @@ def _reconstruct_relative_url(self, environ):
       url += '?' + environ['QUERY_STRING']
     return url
 
+  def reconstruct_hostname(self, port_override=None):
+    """Reconstruct the hostname of a request.
+
+    This is based on the URL reconstruction code in Python PEP 333:
+    http://www.python.org/dev/peps/pep-0333/#url-reconstruction.  Rebuild the
+    hostname from the pieces available in the environment.
+
+    Args:
+      port_override: str, An override for the port on the returned hostname.
+
+    Returns:
+      The hostname portion of the URL from the request, not including the
+      URL scheme.
+    """
+    url = self.server
+    port = port_override or self.port
+    if self.url_scheme == 'https':
+      if port and str(port) != '443':
```

Comment:
Why not just collapse these? `if self.url_scheme == 'https' and port and str(port) != '443'`, and if this is always an int, no reason to convert it to a string